framework/src/multiapps/MultiApp.C

```diff
@@ -253,15 +254,20 @@ MultiApp::MultiApp(const InputParameters & parameters)
     _restore_timer(registerTimedSection("restore", 3, "Restoring MultiApp")),
     _reset_timer(registerTimedSection("resetApp", 3, "Resetting MultiApp"))
 {
-
   if (parameters.isParamSetByUser("cli_args") && parameters.isParamValid("cli_args") &&
       parameters.isParamValid("cli_args_files"))
     paramError("cli_args",
                "'cli_args' and 'cli_args_files' cannot be specified simultaneously in MultiApp ");
 
-  if ((_reset_apps.size() > 0 && _reset_time == std::numeric_limits<Real>::max()) ||
-      (_reset_apps.size() == 0 && _reset_time < std::numeric_limits<Real>::max()))
+  if ((_reset_apps.size() > 0 && _reset_times.size() == 0) ||
+      (_reset_apps.size() == 0 && _reset_times.size() > 0))
     mooseError("reset_time and reset_apps may only be specified together");
+
+  // Check that the reset times are sorted by the user
+  auto sorted_times = _reset_times;
+  std::sort(sorted_times.begin(), sorted_times.end());
+  if (_reset_times.size() && _reset_times != sorted_times)
+    paramError("reset_time", "List of reset times must be sorted in increasing order");
```

Comment:
It doesn't appear as though you are hitting this error case in your test - please add another to hit this for coverage.